src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  // Main Loop:
+  for (const action of actions) {
+    // For the type of 'addProperties':
+    if (action['type'] === 'addProperties') {
+      for (const extraItem in action['extraData']) {
+        state[extraItem] = action['extraData'][extraItem];
+      }
+      // For the type of 'removeProperties':
+    } else if (action['type'] === 'removeProperties') {
+      for (const delItem of action['keysToRemove']) {
+        delete state[delItem];
+      }
+      // For the type of 'clear':
+    } else if (action['type'] === 'clear') {
+      for (const delAllItem in state) {
+        delete state[delAllItem];
+      }
```

Comment:
if you have some logic, that you need to run depending on the value of some variable (action type in this case), will be better to use switch / case